crates/ironrdp-async/src/connector.rs

```diff
@@ -165,6 +148,28 @@ where
                 .await
                 .map_err(|e| ironrdp_connector::custom_err!("write all", e))?;
         }
+
+        if let Some(next_pdu_hint) = sequence.next_pdu_hint() {
```

Comment:
Nit: Maybe we could reduce nesting a bit for a long code block here:
```rust
let next_pdu_hint = if let Some(next_pdu_hint) = sequence.next_pdu_hint() {
    next_pdu_hint
} else {
    break;
};

debug!(
    connector.state = connector.state.name(),
...
```

same suggestion for `crates/ironrdp-blocking/src/connector.rs`

Reply:
Good catch, I adjusted the code accordingly